File: tmp/parse_economic_calendar.js

const fs = require('fs');
const path = require('path');
const { parse } = require('csv-parse/sync'); // Use sync version

function parseCsv() {
    const csvFilePath = path.join(__dirname, 'economic_calendar.csv');
    const csvContent = fs.readFileSync(csvFilePath, 'utf8');

    const records = parse(csvContent, {
        columns: true,
        skip_empty_lines: true,
        trim: true,
    });

    const events = [];
    const noteTxtContent = ''; // As previously determined, note.txt is empty

    for (const record of records) {
        // Skip source line
        if (record.Date && record.Date.startsWith('Source')) {
            continue;
        }

        // Clean up Forecast and Notes fields
        if (record.Forecast === '[]') {
            record.Forecast = '';
        }
        if (record.Notes === '[]') {
            record.Notes = '';
        }

        // Integrate notes from note_txt_content if applicable
        if (noteTxtContent && record.Notes) {
            record.Notes = `${record.Notes.trim()} ${noteTxtContent.trim()}`.trim();
        }

        events.push(record);
    }

    const eventsByDate = {};
    for (const event of events) {
        const dateParts = event.Date.split('-');
        // Convert YY to YYYY if it's 2-digit
        const year = `20${dateParts[2]}`;
        const dateObj = new Date(`${dateParts[1]} ${dateParts[0]}, ${year}`);

        const options = { day: '2-digit', month: 'long', year: 'numeric' };
        const formattedDateKey = dateObj.toLocaleDateString('en-US', options);
        
        // Ensure the month abbreviation is consistent (e.g., Dec not december)
        const anchorDateKey = dateObj.toLocaleDateString('en-US', { day: '2-digit', month: 'short' }).replace(' ', '').toLowerCase();
        const dayName = dateObj.toLocaleDateString('en-US', { weekday: 'short' });

        if (!eventsByDate[formattedDateKey]) {
            eventsByDate[formattedDateKey] = {
                anchor: `day-${anchorDateKey}`,
                day_name: dayName,
                events: []
            };
        }
        eventsByDate[formattedDateKey].events.push(event);
    }

    // For consistent order, sort dates
    const sortedDates = Object.keys(eventsByDate).sort((a, b) => new Date(a) - new Date(b));
    const sortedEventsByDate = {};
    for (const date of sortedDates) {
        sortedEventsByDate[date] = eventsByDate[date];
    }

    console.log(JSON.stringify(sortedEventsByDate, null, 2));
}

parseCsv();

File: tmp/parse_economic_calendar.py

import csv
import json
from datetime import datetime

with open('tmp/economic_calendar.csv', 'r') as f:
    csv_content = f.read()

# Assuming note_txt_content is empty as per previous observation
note_txt_content = ''

events = []
lines = csv_content.strip().split('\n')
reader = csv.reader(lines)
header = [h.strip() for h in next(reader)]

for i, row in enumerate(reader):
    if not row or row[0].strip().startswith('Source'):
        continue

    # Ensure row has enough columns, fill with empty strings if not
    row = [r.strip() for r in row] + [''] * (len(header) - len(row))

    event = dict(zip(header, row))

    # Clean up Forecast and Notes
    if event['Forecast'] == '[]':
        event['Forecast'] = ''
    if event['Notes'] == '[]':
        event['Notes'] = ''

    # Integrate notes from note_txt_content if applicable
    if note_txt_content and 'Notes' in event:
        event['Notes'] = f"{event['Notes'].strip()} {note_txt_content.strip()}".strip()

    events.append(event)

# Group events by date
events_by_date = {}
for event in events:
    date_obj = datetime.strptime(event['Date'], '%d-%b-%y')
    formatted_date_key = date_obj.strftime('%d %B %Y')
    anchor_date_key = date_obj.strftime('%d%b').lower()
    day_name = date_obj.strftime('%a') # Get abbreviated day name (Mon, Tue, etc.)

    if formatted_date_key not in events_by_date:
        events_by_date[formatted_date_key] = {
            'anchor': f'day-{anchor_date_key}',
            'day_name': day_name,
            'events': []
        }
    events_by_date[formatted_date_key]['events'].append(event)

# For consistent order, sort dates
sorted_dates = sorted(events_by_date.keys(), key=lambda x: datetime.strptime(x, '%d %B %Y'))
sorted_events_by_date = {date: events_by_date[date] for date in sorted_dates}

print(json.dumps(sorted_events_by_date, indent=2))
